django-stubs/views/decorators/vary.pyi

```diff
@@ -1,7 +1,10 @@
 from collections.abc import Callable
 from typing import Any, TypeVar
 
+from typing_extensions import ParamSpec
+
+_P = ParamSpec("_P")
 _F = TypeVar("_F", bound=Callable[..., Any])
 
-def vary_on_headers(*headers: Any) -> Callable: ...
+def vary_on_headers(*headers: Callable[_P, _F]) -> Callable[_P, _F]: ...
```

Comment:
This is not correct, the arguments to `headers` are presumably header names as strings?

And the `Callable[]` signature does not preserve the arguments of the wrapped function.

Reply:
Yes you are right! My bad, sorry for that. I dug a bit deeper into mypy docs and found that `functools.wraps` will preserve the types while using the `vary_on_headers` ( as I mentioned in #1340 ). 

```
import functools
from collections.abc import Callable
from typing import Any, TypeVar, cast

_F = TypeVar("_F", bound=Callable[..., Any])

def vary_on_headers(*headers: Any) -> Callable[[_F], _F]:
    def decorator(func: _F) -> _F:
        @functools.wraps(func)
        def wrapper(*args: Any, **kwargs: Any) -> Any:
            return func(*args, **kwargs)
        return cast(_F, wrapper)
    return decorator

def vary_on_cookie(func: _F) -> _F: ...
```
This is a sample implementation, it can be achieved through `Paramspec` as well, without the unsafe `cast()`. But in any case we would need to impliment `vary_on_headers` which fails the flake8 pre-commit hook. 

```
django-stubs/views/decorators/vary.pyi:11:13: Y010 Function body must contain only "..."
django-stubs/views/decorators/vary.pyi:12:9: Y048 Function body should contain exactly one statement
django-stubs/views/decorators/vary.pyi:13:5: Y048 Function body should contain exactly one statement
```